tokio/src/macros/select.rs

```diff
@@ -39,13 +39,13 @@ macro_rules! doc {
         /// 2. Aggregate the `<async expression>`s from each branch, including the
         ///    disabled ones. If the branch is disabled, `<async expression>` is still
         ///    evaluated, but the resulting future is not polled.
-        /// 3. Concurrently await on the results for all remaining `<async expression>`s.
-        /// 4. Once an `<async expression>` returns a value, attempt to apply the value
+        /// 3. If **all** branches are disabled: evaluate the `else` expression if one is provided
+        ///    and return, otherwise panic.
+        /// 4. Concurrently await on the results for all remaining enabled `<async expression>`s.
+        /// 5. Once an `<async expression>` returns a value, attempt to apply the value
         ///    to the provided `<pattern>`, if the pattern matches, evaluate `<handler>`
         ///    and return. If the pattern **does not** match, disable the current branch
         ///    and for the remainder of the current call to `select!`. Continue from step 3.
```

Comment:
Branches become disabled if the pattern doesn't match, which can in turn trigger the else branch.

Reply:
The else branch can also be triggered before anything is polled if all the expressions are disabled via their condition, or if there are no expressions to begin with.

Reply:
We could perhaps list it twice, but I don't think it makes sense to remove the mention of `else` at the end.

Reply:
At present, step 5 about `else` is never reached by following the lifecycle as written. Do you have any suggestions for rewording it to make it more correct?

Reply:
You can say "if all branches are disabled, go to step 6" and keep the step at the end?